worker-workflow/src/main/java/com/github/cafdataprocessing/workflow/ArgumentsManager.java

```diff
@@ -86,7 +94,12 @@ public void addArgumentsToDocument(final List<ArgumentDefinition> argumentDefini
         }
 
         final Map<String, String> arguments = new HashMap<>();
-
+        //Process optional extra failure subfields
+        if (!failureSubfields.isEmpty()) {
+            final List<String> failureFields = failureSubfields.entrySet().stream().map(e -> e.getKey()).collect(Collectors.toList());
+            arguments.putAll(failureSubfields);
+            arguments.put("extraFailureSubfields", gson.toJson(failureFields));
```

Comment:
I sort of feel that the re-use of the arguments structure for this is a bit hacky.  It's probably technically possible that there could be an argument and a failure field with the same name, although admittedly that's unlikely. 
 @andyreidz, what do you think about this?

Reply:
I agree, was there a difficulty with adding a new structure for this? 
This approach is corrupting the ArgumentsManager with additional responsibilities.

Reply:
I added them to this as I seen arguments as the best place for them. They are params passed to the workflow worker with their intent to be solely used within the workflow script. That seems like that made them arguments to the workflow. Also adding them here gave us the rest of their requirements for free, such as being preserved even in the case of poison messages and available for use at all stages of the workflow.

Reply:
This leaves the failure sub files logic spread across two classes, I'm not familiar with this enhancement but it appears to me that this would be better handled in the FailureFieldsManager and not in the ArgumentsManager.